src/Raven.Client/Documents/Session/DocumentSession.Patch.cs

```diff
@@ -99,12 +100,15 @@ public partial class DocumentSession
                 Values = { { "val", val } }
             }));
 
-            _patchInfo = new { DocId = key, Script = $"\"{script}\"", Path = path, Val = val };
+            _patchInfo.Add(new { DocId = key, Script = $"\"{script}\"", Path = path, Val = val });
         }
 
         public override string ToString()
         {
-            return _patchInfo.ToString();
+            var sb = new StringBuilder();
+            foreach (var v in _patchInfo)
```

Comment:
Then maybe in ToString we should iterate through the deferred commands and build the debug output based on them? The advantage is that you will see debug information for all pending commands.
Also _patchInfo is not cleared on SaveChanges